src/LoadTests/TestTailing.cs

```diff
@@ -0,0 +1,368 @@
+                            concurrentWrite.Add((cts, t));
+                        }
+                        else
+                        {
+                            Output.WriteLine("Cancelling concurrent write");
+                            var cts = concurrentWrite.First();
+                            cts.source.Cancel();
+                            cts.source.Dispose();
+                            concurrentWrite.Remove(cts);
+                        }
+                    }
+
+                    if (sw.ElapsedMilliseconds >= nextTimeDelayTx)
+                    {
+                        Output.WriteLine("Adding delay write");
+
+                        nextTimeDelayTx += 1000;
+                        var cts = new CancellationTokenSource();
+                        var t = Task.Run(() =>
+                                AddTransaction(connectionString,
+                                    scheme,
+                                    3050,
+                                    jsonData,
+                                    numberOfMessagesPerAmend,
+                                    cts.Token),
+                            cts.Token);
+                        concurrentDelayedTx.Add((cts, t));
+                    }
+
+                    foreach (var completedWrite in concurrentWrite.Where(x => x.task.IsCompleted).ToList())
+                    {
+                        completedWrite.source.Dispose();
+                        concurrentWrite.Remove(completedWrite);
+                    }
+
+                    if (concurrentWrite.Count < minConcurrent)
+                        up = true;
+                    else if (concurrentWrite.Count > maxConcurrent)
+                        up = false;
+                }
+
+                foreach (var (cancellationTokenSource, _) in concurrentWrite)
+                {
+                    cancellationTokenSource.Cancel();
+                    cancellationTokenSource.Dispose();
+                }
+
+                foreach (var (cancellationTokenSource, _) in concurrentDelayedTx)
+                {
+                    cancellationTokenSource.Cancel();
+                    cancellationTokenSource.Dispose();
+                }
+
+                await Task.WhenAll(concurrentWrite.Select(x => x.task).Where(t => !t.IsCompleted));
+                await Task.WhenAll(concurrentDelayedTx.Select(x => x.task).Where(t => !t.IsCompleted));
+
+                Output.WriteLine("Writes finished");
+
+                var db = new List<long>();
+                ReadAllPage page = await streamStore.ReadAllForwards(Position.Start, 500, false, ct);
+                do
+                {
+                    db.AddRange(page.Messages.Select(x => x.Position));
+                    page = await page.ReadNext(linkedToken.Token);
+                } while (!page.IsEnd);
+
+                db.AddRange(page.Messages.Select(x => x.Position));
+
+                while (true)
```

Comment:
@ArneSchoonvliet Dit dient om te zorgen dat ge zeker klaar zijt met schrijven eh?
Maar kan het niet dat door nen skipped eveznt op het einde dieje while true nooit breakt?

Reply:
exact, en ge hebt gelijk.
ik zal regelen dat hij timebased is zodat dieje niet eeuwig blijft loopen

Reply:
Top!